pkg/cloudprovider/cloud.go

```diff
@@ -83,7 +84,7 @@ type LoadBalancer interface {
 	// if so, what its status is.
 	GetLoadBalancer(name, region string) (status *api.LoadBalancerStatus, exists bool, err error)
 	// EnsureLoadBalancer creates a new load balancer, or updates an existing one. Returns the status of the balancer
-	EnsureLoadBalancer(name, region string, loadBalancerIP net.IP, ports []*api.ServicePort, hosts []string, affinityType api.ServiceAffinity) (*api.LoadBalancerStatus, error)
+	EnsureLoadBalancer(namespacedName types.NamespacedName, name, region string, loadBalancerIP net.IP, ports []*api.ServicePort, hosts []string, affinityType api.ServiceAffinity) (*api.LoadBalancerStatus, error)
```

Comment:
suggest renaming namespacedName to serviceName so that it is clear what the name refers to.  Also it's now ambiguous as to whether we are creating/updating based on namespacedName vs name, so I suggest we expand the comment to specify, and maybe move namespacedName towards the end of the arguments (first place is misleading as to its importance)